source/NVDAObjects/__init__.py

```diff
@@ -813,6 +813,13 @@ def _get_isCurrent(self):
 		"""
 		return False
 
+	def shouldAcceptShowHideCaretEvent(self):

```

Comment:
Just a note that this feels weird being on the base NVDAObject rather than IAccessible, but right now, I understand that's how it has to be because we fire MSAA caret events on the focus object regardless of whether it's IAccessible. I think we should consider restricting these caret events to focus objects that are IAccessible subclasses in future, but that change is probably too risky for this PR. CC @MichaelDCurran for his thoughts.